tools/export/cdt/__init__.py

```diff
@@ -1,18 +1,49 @@
 import re
-
+import os
+import json
+from collections import namedtuple
+from tools.targets import TARGET_MAP
 from os.path import join, exists
 from os import makedirs, remove
 import shutil
 
 from tools.export.makefile import Makefile, GccArm, Armc5, IAR
 
+_eclipse_defs = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)), 'cdt_definitions.json')
+
+with open(_eclipse_defs, 'r') as f:
+    _CONFIGS_OPTIONS = json.load(f)
+
+supported_launches = ['debug', 'program', 'erase']
+
 class Eclipse(Makefile):
     """Generic Eclipse project. Intended to be subclassed by classes that
     specify a type of Makefile.
     """
+    def get_target_config(self, ctx, configuration):
+        """Retrieve info from cdt_definitions.json"""
+        tgt = TARGET_MAP[self.target]
+        defaults = _CONFIGS_OPTIONS['default']
+        eclipse_config = defaults['generic']
+        if configuration in defaults:
+            eclipse_config.update(defaults[configuration])
+
+        target_specific = _CONFIGS_OPTIONS['targets']
+        if tgt.name in target_specific:
+            target_info = target_specific[tgt.name]['generic']
```

Comment:
This is modified, so multiple exports on the online compiler will lead to unintended behavior when they all modify this variable over and over again.
```suggestion
           eclipse_config.update(target_specific[tgt.name]['generic'])
```

If you change ^ and you replace `target_info` with `eclipse_config` 2 lines down, you achieve the same affect without having to modify more dictionaries.